templates/config/controller/deployment.yaml.tpl

```diff
@@ -61,6 +61,7 @@ spec:
             drop:
               - ALL
       terminationGracePeriodSeconds: 10
+      serviceAccountName: ack-{{ .ServicePackageName }}-service-account
```

Comment:
The Helm templates use `{{ .ServiceAccountName }}` here - https://github.com/aws-controllers-k8s/code-generator/blob/main/templates/helm/values.yaml.tpl#L75 . This references an optional parameter in the generator, which defaults to `"default"` - https://github.com/aws-controllers-k8s/code-generator/blob/60b04feaa1e593745eb7beb416a4be7ee6ef69df/cmd/ack-generate/command/release.go#L46

Ideally we would use that same variable here, and update the default value for that optional parameter to be `ack-<service>-service-account`.

Reply:
@RedbackThomson Thanks for pointing this out, I wasn't sure how this was getting picked up in `helm`. I have refactored to pass the same `var` to all necessary processes/flows. Let me know what you think, I had to add this to the `API's` process too, since a variable there is shared with the `controller` process.